### DISCUSSION TITLE
write spin interaction in the spinful_fermion_basis_general

### DISCUSSION THREAD

**Discussion by tpyang** [QuSpin/QuSpin#541] [Q&A]
Hi,

I am trying to work with spin full fermion basis without double degeneracy:
basis_2d=spinful_fermion_basis_general(N_2d,Nf=(Nup,Ndn), double_occupancy=False)

Is there a straightforward way to write the spin interaction term JS_i *S_j? similar as the onsite Hubbard term ["n|n",interaction]?

Thank you very much!
Yang

**Comment by JiahaoYao** [comment]
Do u mean you are working on the spin model? [Here](https://weinbe58.github.io/QuSpin/Examples.html) are other examples for spin models.

**Reply by tpyang** [reply]
Thanks for the quick reply! No, it's a fermion model with hoping and spin interaction, for example the [tJ model](https://en.wikipedia.org/wiki/T-J_model). Currently I only see the pure spin model in the documentation.

**Comment by JiahaoYao** [comment]
Hi @tpyang, this is a pretty good question! You might need to refer to this [basis page](https://weinbe58.github.io/QuSpin/basis.html?highlight=basis#module-quspin.basis).

where lists the correspondence between spin operator and fermi operator.

<img width="827" alt="image" src="https://user-images.githubusercontent.com/20907377/170892170-a87cffb5-0f75-487e-b1d3-7930f55aa7cf.png">


Also, in other reference, the spin operator can also be written as 
<img width="824" alt="image" src="https://user-images.githubusercontent.com/20907377/170892189-5b739b4a-0e92-4b51-aedc-208d946956ce.png">

Let me know whether this helps!



By the way, if you successfully work this out, could you also post your implementation here! Thanks! 
@mgbukov t-J model seems also quite a common model, and also be nice to add in the examples.

**Reply by tpyang** [reply]
Sure, I have implemented it in the triangular lattice with four fermion operator. I will perform some test to make sure it's correct and then post it here.

**Reply by JiahaoYao** [reply]
> Sure, I have implemented it in the triangular lattice with four fermion operator. I will perform some test to make sure it's correct and then post it here. 

Perfect! Feel free to let me know if u encounter any problem!
